ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/sync/fastsync/FastSyncStateStorage.java

```diff
@@ -55,8 +56,23 @@ public FastSyncState loadState(final BlockHeaderFunctions blockHeaderFunctions)
         return FastSyncState.EMPTY_SYNC_STATE;
       }
       final Bytes rlp = Bytes.wrap(Files.toByteArray(pivotBlockHeaderFile));
-      return new FastSyncState(
-          BlockHeader.readFrom(new BytesValueRLPInput(rlp, false), blockHeaderFunctions));
+      final BytesValueRLPInput input = new BytesValueRLPInput(rlp, false);
+
+      if (isNewFormat(input)) {
```

Comment:
isNewFormat is not very a descriptive name

Reply:
That's a better name, will update it